--- freqtrade/persistence/pairlock_middleware.py
+++ freqtrade/persistence/pairlock_middleware.py
@@ -3,7 +3,7 @@
 from collections.abc import Sequence
 from datetime import UTC, datetime

-from sqlalchemy import select
+from sqlalchemy import select, update

 from freqtrade.exchange import timeframe_to_next_date
 from freqtrade.persistence.models import PairLock
@@ -142,10 +142,11 @@
                 PairLock.active.is_(True),
                 PairLock.reason == reason,
             ]
-            locks = PairLock.session.scalars(select(PairLock).filter(*filters)).all()
-            for lock in locks:
-                logger.info(f"Releasing lock for {lock.pair} with reason '{reason}'.")
-                lock.active = False
+            PairLock.session.execute(
+                update(PairLock)
+                .where(*filters)
+                .values(active=False)
+            )
             PairLock.session.commit()
         else:
             # used in backtesting mode; don't show log messages for speed
